Description: Disable breakpad on x32.
 While x32 is not supported, breakpad mistakes it for amd64 (as both share
 the same processor type).  This patch fixes the detection.
 .
 The profiler depends on breakpad, so it has to be disabled as well.
Author: Adam Borowski <kilobyte@angband.pl>

--- iceweasel-34.0.orig/configure.in
+++ iceweasel-34.0/configure.in
@@ -1651,6 +1651,10 @@ Linux)
     *)
         MOZ_ENABLE_PROFILER_SPS=
     esac
+    case "$target" in
+    x86_64-*-linux-gnux32*| x86_64-*-linux-muslx32*)
+        MOZ_ENABLE_PROFILER_SPS=
+    esac
     ;;
 WINNT|Darwin) ;;
 *)
@@ -5929,6 +5932,10 @@ dnl = Breakpad crash reporting (on by de
 dnl ========================================================
 
 case $target in
+x86_64-*-linux-gnux32*| x86_64-*-linux-muslx32*)
+  # x32 is not supported.
+  MOZ_CRASHREPORTER=
+  ;;
 i?86-*-mingw*|x86_64-*-mingw*)
   MOZ_CRASHREPORTER=1
   ;;
--- iceweasel-34.0.orig/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h
+++ iceweasel-34.0/toolkit/crashreporter/google-breakpad/src/third_party/lss/linux_syscall_support.h
@@ -85,7 +85,8 @@
 /* We currently only support x86-32, x86-64, ARM, MIPS, and PPC on Linux.
  * Porting to other related platforms should not be difficult.
  */
-#if (defined(__i386__) || defined(__x86_64__) || defined(__ARM_ARCH_3__) ||   \
+#if (defined(__i386__) || defined(__x86_64__) && defined(__LP64__) \
+     || defined(__ARM_ARCH_3__) ||   \
      defined(__mips__) || defined(__PPC__) || defined(__ARM_EABI__)) \
   && (defined(__linux) || defined(__ANDROID__))
 
